src/sync/partial_sync.cpp

```diff
@@ -33,6 +36,65 @@
 #include <realm/lang_bind_helper.hpp>
 #include <realm/util/scope_exit.hpp>
 
+using namespace std::chrono;
+
+namespace {
+
+    // FIXME: Replace these methods with: https://github.com/realm/realm-core/pull/3259
+
+    /// Turn a system time point value into the 64-bit integer representing ns since the Unix epoch.
+    int64_t ns_since_unix_epoch(const system_clock::time_point& point)
+    {
+        return duration_cast<nanoseconds>(point.time_since_epoch()).count();
+    }
+
+    realm::Timestamp timestamp_now() {
```

Comment:
This could be simplified by letting std::chrono handle the conversions rather than doing the math manually:

    auto now = system_clock::now();
    auto sec = time_point_cast<seconds>(now);
    return {sec.time_since_epoch().count(), duration_cast<nanoseconds>(now - sec).count()};